api/v1/db/aws/aws-operations.js

```diff
@@ -55,26 +53,126 @@ const objectExists = (key, bucket = thisBucket) => new Promise((resolve, reject)
   });
 });
 
+/**
+ * @summary Gets metadata on an object by making a head-object request
+ * @function
+ * @param {string} key Key of the object
+ * @param {string} bucket Bucket where the object exists
+ * @returns {Promise} Promise object representing the response
+ */
+const headObject = (key, bucket = thisBucket) => {
+  const params = { Bucket: bucket, Key: key };
+  return s3.headObject(params).promise();
+};
```

Comment:
As we discussed, if we needed to send `head` request to make sure the object is existed then send `get`/`put`, we could run into the issue that there will be a little delay and may have some incoming `delete`/`post` requests during the delay.